CompilerSource/parser/collect_variables.cpp

```diff
@@ -31,32 +32,119 @@
 
 #include "event_reader/event_parser.h"
 
+#include <JDI/src/System/builtins.h>
 #include <cstdio>
 #include <iostream>
 #include <map>
-#include <string>
+#include <regex>
 #include <sstream>
+#include <string>
 #include <unordered_set>
 
 using namespace std;
 
 extern int global_script_argument_count;
 
 struct scope_ignore {
-  map<string,int> ignore;
+  map<string, int> ignore;
   bool is_with;
-  
-  scope_ignore(scope_ignore*x): is_with(x->is_with) {}
-  scope_ignore(bool x): is_with(x) {}
-  scope_ignore(int x): is_with(x) {}
+
+  scope_ignore(scope_ignore *x) : is_with(x->is_with) {}
+  scope_ignore(bool x) : is_with(x) {}
+  scope_ignore(int x) : is_with(x) {}
 };
 
-#include "object_storage.h"
+#include "../parsing/tokens.h"
 #include "collect_variables.h"
 #include "languages/language_adapter.h"
+#include "object_storage.h"
 
 using enigma::parsing::AST;
 
+std::string GetFullType(enigma::parsing::FullType &ft) {
+  std::string type;
+  std::vector<std::size_t> flags_values = {
+      jdi::builtin_flag__const->value,    jdi::builtin_flag__static->value,       jdi::builtin_flag__volatile->value,
+      jdi::builtin_flag__mutable->value,  jdi::builtin_flag__register->value,     jdi::builtin_flag__inline->value,
+      jdi::builtin_flag__Complex->value,  jdi::builtin_flag__unsigned->value,     jdi::builtin_flag__signed->value,
+      jdi::builtin_flag__short->value,    jdi::builtin_flag__long->value,         jdi::builtin_flag__long_long->value,
+      jdi::builtin_flag__restrict->value, jdi::builtin_typeflag__override->value, jdi::builtin_typeflag__final->value};
+
+  std::vector<std::size_t> flags_masks = {
+      jdi::builtin_flag__const->mask,    jdi::builtin_flag__static->mask,       jdi::builtin_flag__volatile->mask,
+      jdi::builtin_flag__mutable->mask,  jdi::builtin_flag__register->mask,     jdi::builtin_flag__inline->mask,
+      jdi::builtin_flag__Complex->mask,  jdi::builtin_flag__unsigned->mask,     jdi::builtin_flag__signed->mask,
+      jdi::builtin_flag__short->mask,    jdi::builtin_flag__long->mask,         jdi::builtin_flag__long_long->mask,
+      jdi::builtin_flag__restrict->mask, jdi::builtin_typeflag__override->mask, jdi::builtin_typeflag__final->mask};
+
+  std::vector<std::string> flags_names = {"const",  "static",    "volatile", "mutable",  "register",
+                                          "inline", "complex",   "unsigned", "signed",   "short",
+                                          "long",   "long long", "restrict", "override", "final"};
```

Comment:
what is the purpose of having these separate instead of an dict map or associative array?

I would at least assert that they are the same sizes and such.